Library/Homebrew/update_migrator.rb

```diff
@@ -0,0 +1,39 @@
+require "update_migrator/cache_entries_to_double_dashes"
+require "update_migrator/cache_entries_to_symlinks"
+require "update_migrator/legacy_cache"
+require "update_migrator/legacy_keg_symlinks"
+require "update_migrator/legacy_repository"
+
+module UpdateMigrator
+  module_function
+
+  def formula_resources(formula)
+    specs = [formula.stable, formula.devel, formula.head].compact
+
+    [*formula.bottle&.resource] + specs.flat_map do |spec|
+      [
+        spec,
+        *spec.resources.values,
+        *spec.patches.select(&:external?).map(&:resource),
```

Comment:
How about `compact` and/or `flatten` instead of `*` for these?

Reply:
`*` is more concise here, I think.